hms-app/src/components/IdeaCardFoldable.tsx

```diff
@@ -50,136 +74,191 @@ export default function IdeaCardFoldable(props: IProps) {
     initialItem: -1,
   })
 
-  const { idea, type } = props
+  const [ideaData, setIdeaData] = useState({
+    errorIdeaData: false,
+    isLoadingIdeaData: true,
+    id: 'string',
+    owner: [],
+    hackathon: [],
+    participants: [],
+    title: 'string',
+    description: 'string',
+    problem: 'string',
+    goal: 'string',
+    requiredSkills: [],
+    category: [],
+    creationDate: 'string',
+  })
+
+  const {
+    errorIdeaData,
+    isLoadingIdeaData,
+    id,
+    owner,
+    hackathon,
+    participants,
+    title,
+    description,
+    problem,
+    goal,
+    requiredSkills,
+    category,
+    creationDate,
+  }: Idea = ideaData
+
+  const loadIdeaDetails = () => {
+    getIdeaDetails(ideaPreview.id.toString()).then(
+      (data) => {
+        setIdeaData({
+          id: data.id,
+          owner: data.owner,
+          hackathon: data.hackathon,
+          participants: data.participants,
+          title: data.title,
+          description: data.description,
+          problem: data.problem,
+          goal: data.goal,
+          requiredSkills: data.requiredSkills,
+          category: data.category,
+          creationDate: data.creationDate,
+          errorIdeaData: false,
+          isLoadingIdeaData: false,
+        })
+      },
+      () => {
+        setIdeaData({
+          ...ideaData,
+          errorIdeaData: true,
+          isLoadingIdeaData: false,
+        })
+      }
+    )
+  }
+
+  const { ideaPreview, type } = props
 
   const MAX_TITLE_LENGTH = 45
   const MAX_DESCRIPTION_LENGTH = type === 'voting' ? 200 : 245
 
-  const skillBadges = idea.skills.map((skill) => (
-    <Badge color={theme.colorScheme === 'dark' ? 'dark' : 'gray'} key={skill}>
-      {skill}
-    </Badge>
-  ))
-
-  const participantAvatars = idea.participants.map((participant) => (
-    <Avatar key={participant.name} src={participant.avatar} />
-  ))
+  useEffect(() => {
+    loadIdeaDetails()
+  }, [])
```

Comment:
technically if the `ideaPreview` variable changed, this should reload.  But I don't know if react will reuse components like that or if it just makes new ones :thinking:.  That is to say this is most likely fine with just `[]`, more I'm just thinking out loud... or in text on the internet